### PR TITLE
merge danny branch with complete recoding and partial QA

### PR DESCRIPTION
**Pull request by dcolombara**
- confirmed that load_stage.bir_wa.R works … appends 2003-2017 data and recodes to create vars needed for APDE

- QA script is functional, but incomplete

### CODE REVIEW

File: ETL/birth/stage/load_stage.bir_wa.R [hunk 1]
```diff
@@ -644,40 +658,250 @@ bir_combined <- setDT(bind_rows(bir_2017_20xx, bir_2003_2016))
...
   tbl_id_2003_20xx <- DBI::Id(schema = table_config_stage_bir_wa$schema, 
                               table = table_config_stage_bir_wa$table)
-  dbWriteTable(db_apde, tbl_id_2003_20xx, value = as.data.frame(bir_combined),
-               overwrite = T, append = F,
+  dbWriteTable(db_apde, 
+               tbl_id_2003_20xx, 
+               value = as.data.frame(bir_combined),
+               overwrite = T, 
+               append = F,
                field.types = unlist(table_config_stage_bir_wa$vars))
```
Comment (alastairmatheson) [inline ETL/birth/stage/load_stage.bir_wa.R]: I think this needs to be field.types = c(unlist(table_config_stage_bir_wa$vars), unlist(table_config_stage_bir_wa$recodes)), right?

File: ETL/birth/stage/load_stage.bir_wa.R [hunk 2]
```diff
@@ -644,40 +658,250 @@ bir_combined <- setDT(bind_rows(bir_2017_20xx, bir_2003_2016))
...
+    # Warning: max integer size is 2147483648 so this code will break in ~140 years
+    bir_combined[date_of_birth_year == 2012, 
+                 birth_cert_encrypt := as.integer(paste0(2012L, birth_cert_encrypt))]   
+  
+#### ________________________________________________________----    
+####                     RECODING                            ----    
+#### ________________________________________________________----    
+    
+#### FREE UP MEMORY ----
+  gc()
```
Comment (alastairmatheson) [inline ETL/birth/stage/load_stage.bir_wa.R]: I've read in various places that using gc() is a placebo but I still use it sometimes.

File: ETL/birth/stage/load_stage.bir_wa.R [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+  # Purpose: COMBINE BIRTH DATA FROM THE BEDROCK (2003-2016) AND WHALES (2017-) SYSTEMS
+  #          Clean and recode variables needed for APDE analyses
+  
+  # Recode groups of variables where <group_name>1-<group_nameX> has been replaced
+  #     by specific flags rather than codes
+  # Standardize missing to be NULL
+  # Standardize country/county/state/city codes (not needed as there is literal and code now)
 
 #### Set-up environment ----
 rm(list=ls())
```
Comment (alastairmatheson) [inline ETL/birth/stage/load_stage.bir_wa.R]: I'd like for this to eventually be drive from the master script, in which case clearing the environment and calling in libraries would come out of here. It's fine for now.

File: ETL/birth/stage/qa_stage.bir_wa.R [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    options(warning.length = 8170) # get lengthy warnings, needed for SQL
+    setwd("C:/temp")
+
+## Load staged from SQL ----
+    table_config_stage_bir_wa <- yaml::yaml.load(getURL(
+      "https://raw.githubusercontent.com/PHSKC-APDE/DOHdata/master/ETL/birth/stage/create_stage.bir_wa.yaml"))
+    
+    query.varlist <- paste(c(names(table_config_stage_bir_wa$vars), names(table_config_stage_bir_wa$recodes)), collapse = ", ")
+    
+    query.string <- paste("SELECT", query.varlist, "FROM stage.bir_wa")
```
Comment (alastairmatheson) [inline ETL/birth/stage/qa_stage.bir_wa.R]: Would recommend switching this to glue_sql but not the end of the world to keep this as is.

File: ETL/birth/stage/load_stage.bir_wa.R (hunk 1 above)
Comment (dcolombara) [reply ETL/birth/stage/load_stage.bir_wa.R]: right! I just fixed this.

File: ETL/birth/stage/load_stage.bir_wa.R (hunk 2 above)
Comment (dcolombara) [reply ETL/birth/stage/load_stage.bir_wa.R]: I'm still on the fence about this. I've also seen discussion saying that it's useless, but my experience tells me it matters. Then I remember Noel Weiss saying the plural of experience isn't evidence. I'll leave it for now so I can sleep better at night.

File: ETL/birth/stage/load_stage.bir_wa.R (hunk 3 above)
Comment (dcolombara) [reply ETL/birth/stage/load_stage.bir_wa.R]: okay. When it's all functional, let's tweak things to run everything from the master.

**Pull request merged**